libs/blocks/countdown-timer/countdown-timer.js

```diff
@@ -0,0 +1,238 @@
+import { createTag } from '../../utils/utils.js';
+
+export class CountdownTimer extends HTMLElement {
+  #daysLeft = '';
+
+  #hoursLeft = '';
+
+  #minutesLeft = '';
+
+  #isVisible = false;
+
+  #timeWords = [];
+
+  #timeRangesEpoch = [];
+
+  constructor() {
+    super();
+    this.attachShadow({ mode: 'open' });
+
+    this.label = '';
+    this.dayshoursmins = '';
+    this.timeranges = '';
+  }
+
+  static get observedAttributes() {
+    return ['label', 'dayshoursmins', 'timeranges'];
+  }
+
+  attributeChangedCallback(name, oldValue, newValue) {
+    this[name] = newValue;
+  }
+
+  connectedCallback() {
+    this.timeWords = this.dayshoursmins.split(' ');
+    this.timeRangesEpoch = this.timeranges.split(',');
+    this.countdownStart();
+  }
+
+  disconnectedCallback() {
+    clearInterval(this.countdownInterval);
+  }
+
+  countdownCompleted() {
+    this.isVisible = false;
+    clearInterval(this.countdownInterval);
+    this.update();
+  }
+
+  countdownStart() {
+    const oneMinuteInMs = 60000;
+    this.countdownUpdate();
+    this.countdownInterval = setInterval(() => { this.countdownUpdate(); }, oneMinuteInMs);
+  }
+
+  countdownUpdate() {
+    const currentTime = Date.now();
+
+    for (let i = 0; i < this.timeRangesEpoch.length; i += 2) {
+      const startTime = this.timeRangesEpoch[i];
+      const endTime = this.timeRangesEpoch[i + 1];
+
+      if (currentTime >= startTime && currentTime <= endTime) {
+        this.isVisible = true;
+        const diffTime = endTime - currentTime;
+        this.daysLeft = Math.floor(diffTime / (1000 * 60 * 60 * 24));
+        this.hoursLeft = Math.floor((diffTime % (1000 * 60 * 60 * 24)) / (1000 * 60 * 60));
+        this.minutesLeft = Math.floor((diffTime % (1000 * 60 * 60)) / (1000 * 60));
```

Comment:
Maybe you can set the attributes here so that they remain consistent with the internal variables.